$OpenBSD: patch-get_iplayer,v 1.23 2009/11/09 19:47:42 sthen Exp $

- default to using id3tag not id3v2
- follow hier(7) not Linux FHS for options file location

--- get_iplayer.orig	Sun Nov  8 14:49:11 2009
+++ get_iplayer	Mon Nov  9 19:47:00 2009
@@ -311,17 +311,14 @@ if ( defined $ENV{GETIPLAYERUSERPREFS} && $ENV{GETIPLA
 # Options on unix-like systems
 } elsif ( defined $ENV{HOME} ) {
 	$profile_dir = $opt_pre->{profiledir} || $ENV{HOME}.'/.get_iplayer';
-	$optfile_system = '/var/lib/get_iplayer/options';
-	if ( -f '/etc/get_iplayer/options' ) {
-		logger "WARNING: System-wide options in /etc/get_iplayer/options are now ignored, please use /var/lib/get_iplayer/options instead\n";
-	}
+	$optfile_system = '${SYSCONFDIR}/get_iplayer/options';
 }
 # Make profile dir if it doesnt exist
 mkpath $profile_dir if ! -d $profile_dir;
 
 
 # get list of additional user plugins and load plugin
-my $plugin_dir_system = '/usr/share/get_iplayer/plugins';
+my $plugin_dir_system = '${TRUEPREFIX}/share/get_iplayer/plugins';
 my $plugin_dir_user = "$profile_dir/plugins";
 for my $plugin_dir ( ( $plugin_dir_user, $plugin_dir_system ) ) {
 	if ( opendir( DIR, $plugin_dir ) ) {
@@ -442,6 +439,7 @@ if ( ! keys %plugin_files ) {
 	$opt->{pluginsupdate} = 1;
 }
 # Update this script if required
+$opt->{packagemanager}='disable';
 update_script() if $opt->{update} || $opt->{pluginsupdate};
 
 
@@ -638,7 +636,7 @@ sub find_matches {
 	delete $binopts->{vlc};
 	push @{ $binopts->{vlc} }, '-vv' if $opt->{debug};
 
-	$bin->{id3v2}		= $opt->{id3v2} || 'id3v2';
+	$bin->{id3v2}		= $opt->{id3v2} || 'id3tag';
 	$bin->{atomicparsley}	= $opt->{atomicparsley} || 'AtomicParsley';
 
 	$bin->{tee}		= 'tee';
@@ -3638,7 +3636,7 @@ sub tag_file {
 		my ( $id3_name, $id3_episode, $id3_desc, $id3_channel ) = ( $prog->{name}, $prog->{episode}, $prog->{desc}, $prog->{channel} );
 		s|"|\\"|g for ($id3_name, $id3_episode, $id3_desc, $id3_channel);
 		# Only tag if the required tool exists
-		if ( main::exists_in_path('id3v2') ) {
+		if ( main::exists_in_path('id3tag') ) {
 			main::logger "INFO: id3 tagging $prog->{ext} file\n";
 			my @cmd = (
 				$bin->{id3v2}, 
@@ -5352,7 +5350,7 @@ sub optional_list_entry_format {
 # Returns the modes to try for this prog type
 sub modelist {
 	my $prog = shift;
-	my $mlist = $opt->{tvmode} || $opt->{modes};
+	my $mlist = $opt->{tvmode} || $opt->{vmode} || $opt->{modes};
 	
 	# Defaults
 	if ( ! $mlist ) {
